docs/blog/2025-01-20-linking-ts-in-monorepos.md

```diff
@@ -0,0 +1,666 @@
+
+## Which One Should I Choose?
+
+Here are some thoughts on which approach to use.
+
+**TypeScript Path Aliases: A Simple Option**
+
+TypeScript path aliases have been a reliable way to manage package resolution, particularly before package managers introduced the workspaces feature. They're straightforward to set up, requiring only a global `tsconfig.json` without additional. However, there are limitations to consider as they require additional bundling support/alias resolvers at runtime and might come with some performance degradation in large workspaces.
+
+{% callout type="deepdive" title="Isn't Nx using TS Path Aliases?" %}
+Yes and no. Nx has been around before package managers introduced workspaces. As a result, the default setup in Nx traditionally leveraged a root-level `tsconfig.base.json` containing path aliases to link packages within the monorepo.
+
+That said, Nx can also be used in combination with the NPM/Yarn/PNPM/Bun workspaces, as [shown here](/recipes/adopting-nx/adding-to-monorepo). This led to two distinct setups for monorepos with Nx: one using TypeScript path aliases and the other leveraging workspaces. To address the confusion this created, the Nx team has spent the last year enhancing Nx to unify these approaches. The goal is to align with "the platform" by adopting and promoting the workspaces, while updating Nx plugins to fully support it, preserving the developer experience (DX) benefits Nx users have come to love.
+
+If you're currently using the TypeScript path aliases approach, there's no need to worry. The Nx team is working on comprehensive documentation and semi-automated tools to help with migration. Additionally, it's possible to migrate manually and even incrementally, allowing you to adopt the workspaces at your own pace.
+{% /callout %}
+
+**Workspaces: The Recommended Approach**
+
+With widespread support in modern package managers (NPM, PNPM, Yarn, Bun), the workspaces feature has become the preferred method for managing package resolution. It aligns closely with the Node.js platform, leveraging native package resolution mechanisms that also work at runtime. This eliminates the portability issues inherent to TypeScript path aliases.
+
+When combined with **TypeScript project references**, this method becomes even more powerful. Workspaces handles package linking s.t. Node can resolve them properly, while TypeScript project references optimize type-checking and enable incremental builds (for TypeScript). Together, they improve performance, reduce memory usage, and simplify dependency management in larger workspaces. This combination is the recommended way to structure and manage TypeScript packages in a monorepo.
```

Comment:
```suggestion
When combined with **TypeScript project references**, this method becomes even more powerful. Workspaces handles package linking such that Node can resolve them properly, while TypeScript project references optimize type-checking and enable incremental builds (for TypeScript). Together, they improve performance, reduce memory usage, and simplify dependency management in larger workspaces. This combination is the recommended way to structure and manage TypeScript packages in a monorepo.
```

Reply:
This is not a standard/ubiquitous abbreviation in English (not like e.g. or i.e.) so should be written out to avoid confusion